### DISCUSSION TITLE
针对 intel 核显和独显更快的方案：使用IPEX-LLM

### DISCUSSION THREAD

**Discussion by circlerain** [SakuraLLM/SakuraLLM#129] [Ideas]
详见：https://github.com/intel-analytics/ipex-llm/

在我的电脑上已经运行成功，用的是 A770 ，测试无论是前加载还是生成速度都明显要快于 vulkan ：
![图片](https://github.com/user-attachments/assets/5de32c7d-9b18-4831-b616-0000bf263e25)

搭建参考：

按照 ipex-llm 教程[使用 IPEX-LLM 在 Intel GPU 上运行 llama.cpp](https://github.com/intel-analytics/ipex-llm/blob/main/docs/mddocs/Quickstart/llama_cpp_quickstart.zh-CN.md)搭建好环境。

进入该环境，下载llama cpp sycl版本放到llama一键包快速启动文件夹下：

![图片](https://github.com/user-attachments/assets/d827c1ca-54de-4764-9dca-08c774ff5ab9)

稍微改动下 common.bat:

```
common.bat:
...
echo 准备启动Sakura服务器...
@echo on
.\sycl llama文件夹名\server.exe -m .\%model.name%.gguf -c 2048 -ngl %ngl% -a %model.name% --host 0.0.0.0 
@echo off

```
然后**进入conda环境中执行bat**。

![图片](https://github.com/user-attachments/assets/242e809b-45d2-434a-a53b-c9a9b47376ab)

不知道wiki可否加入该教程，我搜索了一下目前本项目还没有针对 ipex-llm 的讨论。

_不再让大模型是NVIDIA的专有领域！_

**Comment by soloopooo** [comment]
忘了切号用了别人的github号了，在此对sakurallm的各位开发者深感抱歉，有回应请call我即可，谢谢

**Comment by kurikomoe** [comment]
感觉可以建一个 awesome sakurallm 的 repo 专门整理收集相应的教程和资源。现在关于大模型机翻的教程太分散了。

**Comment by yjbsl** [comment]
我在a770上试了下sakurallm，从第二个输出开始，模型可能会输出乱码。相反，用同样的模型，在4080上就没有这些问题。ipex-llm的issue里也有人遇到了同样的问题。

**Reply by theskanthunt42** [reply]
说到这个我记得我之前乱码的时候看输出他会说1 more token needed to output啥的，我觉得是VRAM不足的问题稍微想办法降低显存占用之后貌似又没事了，我用的是B580
